packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -412,7 +415,10 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
           )
         ),
         new SizedBox(
-          height: kBottomNavigationBarHeight,
+          height: math.max(
```

Comment:
Can you add some tests verifying that this does the right thing?

Reply:
Duh.  Yes, sorry, just got over excited about getting it in.

Reply:
I know that feeling :)

Reply:
Test added.